### PR TITLE
sl-148

### PR DESCRIPTION
**Pull request by RS882**
feat: Create a separate endpoint to display offers for which the user will place a bid and which are in the statuses “auction_started, qualification, completed, canceled, blocked by administrator.”

### CODE REVIEW

File: src/main/java/de/ait/secondlife/controllers/OfferController.java [hunk 1]
```diff
@@ -406,6 +412,57 @@ public ResponseEntity<OfferResponseWithPaginationDto> search(
...
+            summary = "Get all offers by user id in which the user has placed a bid",
+            description = "Receiving all offers by user id available in the database with pagination" +
+                    " in which the user has placed and the offers have statuses : " +
+                    "AUCTION_STARTED, QUALIFICATION, COMPLETED, CANCELED, BLOCKED_BY_ADMIN"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = OfferResponseWithPaginationDto.class))),
+    })
+    public ResponseEntity<OfferResponseWithPaginationDto> getAllByUserBidByUserId(
```
Comment (ychepel) [inline src/main/java/de/ait/secondlife/controllers/OfferController.java:426]: 1) is this for some other task? not for https://cohort38.atlassian.net/browse/SL-148 ?
2) names of methods in the controller, service, and repository are too complicated. could you write analogue in russian? I think we can simplify it

File: src/main/java/de/ait/secondlife/services/OfferServiceImpl.java [hunk 2]
```diff
@@ -327,9 +350,24 @@ private void checkOfferIfIsFree(BigDecimal startPrice, BigDecimal winBin) {
...
         if (id == null) throw new IdIsNullException();
         return offerRepository.existsByIdAndIsActiveTrue(id);
     }
+
+    private void checkUserCredentials(Long id) {
+        try {
+            User user = userService.getAuthenticatedUser();
+            if (!user.getId().equals(id))
+                if (!user.getAuthorities().contains(Role.ROLE_ADMIN))
+                    throw new NoRightToChangeException("User  is not authorised for this operation");
```
Comment (ychepel) [inline src/main/java/de/ait/secondlife/services/OfferServiceImpl.java:369]: why do you use this type of exception here? no modification operations are expected

**Review by ychepel — COMMENTED** [review]
needs to be discussed. until I understood what exactly the problem was solved here and what endpoint will return to the user all his/her offers in different statuses?

File: src/main/java/de/ait/secondlife/controllers/OfferController.java [hunk 3]
```diff
@@ -40,6 +42,12 @@ public class OfferController {
     private final String PAGE_VALUE = "0";
     private final String SIZE_VALUE = "10";
     private final String SORT_BY = "createdAt";
+    private Set<OfferStatus> STATUSES_FOR_BID_SEARCH = Set.of(
```
Comment (ychepel) [inline src/main/java/de/ait/secondlife/controllers/OfferController.java:45]: move to service

File: src/main/java/de/ait/secondlife/controllers/OfferController.java (hunk 1 above)
Comment (ychepel) [inline src/main/java/de/ait/secondlife/controllers/OfferController.java:426]: rename to getUserAuctionParticipation()

File: src/main/java/de/ait/secondlife/controllers/OfferController.java [hunk 4]
```diff
@@ -406,6 +412,57 @@ public ResponseEntity<OfferResponseWithPaginationDto> search(
         return ResponseEntity.ok(service.searchOffers(getPageable(page, size, sortBy, isAsc), locationId, pattern));
     }
 
+    @GetMapping("/user/{id}/bid")
```
Comment (ychepel) [inline src/main/java/de/ait/secondlife/controllers/OfferController.java:415]: change to /participations/user/{id}

File: src/main/java/de/ait/secondlife/controllers/OfferController.java (hunk 3 above)
Comment (RS882) [reply src/main/java/de/ait/secondlife/controllers/OfferController.java:45]: fixed

File: src/main/java/de/ait/secondlife/controllers/OfferController.java (hunk 1 above)
Comment (RS882) [reply src/main/java/de/ait/secondlife/controllers/OfferController.java:426]: fixed

File: src/main/java/de/ait/secondlife/controllers/OfferController.java (hunk 4 above)
Comment (RS882) [reply src/main/java/de/ait/secondlife/controllers/OfferController.java:415]: fixed

**Review by ychepel — APPROVED** [review]

**Pull request merged**